test/case/misc/support_collect/test.py

```diff
@@ -0,0 +1,185 @@
+        has_gpg = (result.returncode == 0)
+        if has_gpg:
+            print("GPG is available on target - will test encryption")
+        else:
+            print("GPG not available on target - skipping encryption tests")
+
+    with test.step("Run support collect with --work-dir and short log tail"):
+        # Create temporary file for output
+        with tempfile.NamedTemporaryFile(suffix=".tar.gz", delete=False) as tmp:
+            output_file = tmp.name
+
+        # Use /tmp as work-dir to test the --work-dir option
+        # Run support collect via SSH with short log tail for testing
+        # Capture stdout (the tarball) to file
+        # Note: timeout is generous to handle systems with many network ports
+        # (ethtool collection scales with number of interfaces)
+        with open(output_file, 'wb') as f:
+            result = tgtssh.run("support --work-dir /tmp collect --log-sec 2",
+                                stdout=f,
+                                stderr=subprocess.PIPE,
+                                timeout=120)
+
+        if result.returncode != 0:
+            stderr_output = result.stderr.decode('utf-8') if result.stderr else ""
+            print(f"support collect failed with return code {result.returncode}")
+            print(f"stderr: {stderr_output}")
+            raise Exception("support collect command failed")
+
+    with test.step("Verify tarball was created and is valid"):
+        if not os.path.exists(output_file):
+            raise Exception(f"Output file {output_file} was not created")
+
+        file_size = os.path.getsize(output_file)
+        if file_size == 0:
+            raise Exception("Output tarball is empty")
+
+        print(f"Tarball created: {file_size} bytes")
+
+        # Verify it's a valid tar.gz
+        try:
+            with tarfile.open(output_file, 'r:gz') as tar:
+                members = tar.getnames()
+                print(f"Tarball contains {len(members)} files/directories")
+
+                # Verify some expected files exist
+                expected_files = [
+                    'collection.log',
+                    'operational-config.json',
+                    'system/dmesg.txt',
+                    'system/meminfo.txt',
+                    'network/ip/addr.json'
+                ]
+
+                root_dir = members[0] if members else None
+                for expected in expected_files:
+                    full_path = f"{root_dir}/{expected}" if root_dir else expected
+                    if full_path not in members:
+                        print(f"Warning: Expected file '{expected}' not found in tarball")
+                    else:
+                        print(f"Found: {expected}")
+
+        except tarfile.TarError as e:
+            raise Exception(f"Invalid tarball: {e}")
+
+        finally:
+            # Clean up
+            if os.path.exists(output_file):
+                os.remove(output_file)
+
+    if has_gpg:
```

Comment:
```suggestion
    if not has_gpg:
      test.succeed()
```
Just for future endeavors: this pattern should work, IIRC. Avoids having to indent the whole following block :nerd_face: